tests/functional/testUtapiClient.js

```diff
@@ -1,37 +1,94 @@
 import assert from 'assert';
 import { mapSeries, series } from 'async';
 import UtapiClient from '../../src/lib/UtapiClient';
-import MemoryBackend from '../../src/lib/backend/Memory';
 import Datastore from '../../src/lib/Datastore';
-import { getBucketCounters, getMetricFromKey } from '../../src/lib/schema';
+import redisClient from '../../src/utils/redisClient';
+import { Logger } from 'werelogs';
+import { getBucketCounters, getMetricFromStateKey,
+    getBucketStateKeys } from '../../src/lib/schema';
 const testBucket = 'foo';
-const memBackend = new MemoryBackend();
 const datastore = new Datastore();
 const utapiClient = new UtapiClient();
 const reqUid = 'foo';
-datastore.setClient(memBackend);
+const redis = redisClient({ host: '127.0.0.1', port: 6379 }, Logger);
+
+datastore.setClient(redis);
 utapiClient.setDataStore(datastore);
 
-function _assertCounters(bucket, cb) {
+function _assertCounters(bucket, valueObj, cb) {
     const counters = getBucketCounters(testBucket);
     return mapSeries(counters, (item, next) =>
-        memBackend.get(item, (err, res) => {
+        datastore.get(item, (err, res) => {
             if (err) {
                 return next(err);
             }
-            const metric = getMetricFromKey(item, bucket)
-                .replace(':counter', '');
-            assert.equal(res, 0, `${metric} must be 0`);
+            let metric = getMetricFromStateKey(item, bucket);
+            metric = metric.replace(':counter', '');
+            assert.equal(res, valueObj[metric],
```

Comment:
Please change this to use `assert.strictEqual` here. Your tests are actually comparing a string to a number.
Pasting a console.log here
```
res [ '9' ] valueObj[metric] 9
res [ '1' ] valueObj[metric] 1
```